front-end/src/Views/EditProjectPage.js

```diff
@@ -0,0 +1,395 @@
+import MenuItem from '@material-ui/core/MenuItem';
+import "../index.scss";
+import Select from '@material-ui/core/Select';
+import FormControl from '@material-ui/core/FormControl';
+// import { theme } from "../Common/ColorTheme";
+import AlertTypes from '../Common/AlertTypes.json';
+import { getAccessToken, getUserId } from '../Components/Auth/Authenticate';
+import { format } from "date-fns";
+import UserNavbar from "../Components/TopBar/UserNavbar";
+
+const Messages = {
+  emptyFieldError: "Please Fill All Areas!",
+  titleEmpty: "Project title can not be empty!",
+  projectEditFail: "Project Edit Failed. Please try again.",
+  projectEditSuccess: "Project Succussfully Edited."
+}
+const projectTypes = {
+  conference: "conference",
+  journal: "journal",
+  institution: "instutution"
+}
+const eventTypes = {
+  conference: "academic conference",
+  journal: "journal submission",
+  institution: "funded project"
+}
+const projectStates = {
+  seekingForCollab: "seeking for collaborators",
+  openForCollab: "open for collaborators",
+  inProg: "in progress",
+  done: "done"
+}
+
+const leftDiv = {
+  margin: "auto",
+  width: "50%",
+  textAlign: "center",
+  minWidth: "500px",
+  display: "inline-block",
+  height: "420px",
+  verticalAlign: "middle",
+  lineHeight: "1"
+
+}
+const rightDiv = {
+  margin: "auto",
+  width: "50%",
+  textAlign: "center",
+  minWidth: "500px",
+  display: "inline-block",
+  height: "420px",
+  verticalAlign: "middle",
+  lineHeight: "1"
+
+}
+const dropdownMenuStyle = {
+  minWidth: "150px",
+  marginLeft: "12px",
+  marginTop: "10px"
+}
+
+const width = {
+  width: "60%",
+  minWidth: "450px"
+
+}
+const FormWrapper = styled(Box)({
+  backgroundColor: "#E0E0E0",
+  border: 0,
+  borderRadius: 3,
+  boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
+  width: "60%",
+  padding: "15px",
+  margin: "10px auto auto auto",
+  '& .MuiTextField-root': {
+
+    margin: "10px",
+    width: "30%",
+    minWidth: "250px"
+  }
+});
+
+
+const Container = styled(Box)({
+  background: "white",
+  border: 0,
+  borderRadius: 3,
+  boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
+  color: 'white',
+  height: "100vh",
+  width: "100%",
+
+  '& .MuiTextField-root': {
+
+    margin: "10px",
+    width: "30%",
+    minWidth: "250px"
+  }
+});
+
+export default class EditProjectPage extends Component {
+  constructor(props) {
+    super(props);
+    this.SnackbarRef = React.createRef();
+    this.state = {
+      success: null,
+      message: "",
+      messageType: "",
+      projectTitle: "Title",
+      projectType: "",
+      projectDescription: "",
+      projectRequirements: "",
+      dueDate: format(new Date(), 'yyyy-MM-dd'),
+      projectState: "",
+      // collaborator: "",
+      collaborators: [this.getSelfProfile()],  //Other members to be added later TODO
+      isPublic: true,
+      events: [],
+      event : ""
+    }
+  };
+  handleDateChange = (date) => {
```

Comment:
We need to add ComponentDidMount method, and inside it you need to fetch the project with get. 
You can see an example of that in ProjectPage.js ^^